storage/remote/write_handler.go

```diff
@@ -140,7 +151,7 @@ func (h *writeHandler) write(ctx context.Context, req *prompb.WriteRequest) (err
 		for _, ep := range ts.Exemplars {
 			e := exemplarProtoToExemplar(&b, ep)
 
-			_, exemplarErr = app.AppendExemplar(0, labels, e)
+			_, exemplarErr = timeLimitApp.AppendExemplar(0, labels, e)
```

Comment:
we could need to override `AppendExemplar` for timeLimitAppender, but let's wait for the feature's GA, the test should fail by then anyway.